### PR TITLE
Check state before deleting or modifying

### LINKED ISSUE #36: We are able to delete a maintenance that is running
@hdiogenes @ajoaoff @rmotitsuki 
It has been tested the API call /maintenance/{mw_id} on DELETE with the idea to execute a delete over a maintenance task that is running looking for a failing result as suppose to. Unfortunately, this task succeeded, meaning that it is possible to delete a running maintenance task.
The test code used is as follow:
```
# Setup maintenance window data
        start = datetime.now() + timedelta(seconds=60)
        end = start + timedelta(seconds=60)
        payload = {
            "description": "mw for test 050",
            "start": start.strftime(TIME_FMT),
            "end": end.strftime(TIME_FMT),
            "items": [
                "00:00:00:00:00:00:00:02"
            ]
        }

        # Request maintenance schema creation, and extract the mw_id
        json_data = self.request_maintenance(payload, False)
        mw_id = json_data["mw_id"]

        time.sleep(10)

        # Delete maintenance mw_id
        api_url = KYTOS_API + '/maintenance/' + mw_id
        delete_response = requests.delete(api_url)
        assert delete_response.status_code == 200

        # Get maintenance schemas (it should be empty)
        json_data = self.get_maintenance_schema()
        assert json_data == []
```

### PR DESCRIPTION
**Pull request by ajoaoff**
Fix #36

### :bookmark_tabs: Description of the Change

Update and delete maintenance now checks the state of the maintenance. If it is
running, the procedure is cancelled and an error code is returned.

### :computer: Verification Process

Unit test was coded and all unit tests passed

### :page_facing_up: Release Notes

- Fix an issue where updates and deletes were allowed when the maintenance is running.

### CODE REVIEW

**Comment by ajoaoff** [comment]
@ArturoQuintana the problem was not introduced by this PR, but by PR #30. It has been fixed in PR #42.

File: main.py [hunk 1]
```diff
@@ -104,6 +107,9 @@ def remove_mw(self, mw_id):
         except KeyError:
             return jsonify({'response': f'Maintenance with id {mw_id} not '
                                         f'found'}), 404
+        if maintenance.status == Status.RUNNING:
+            return jsonify({'response': f'Deleting a running maintenance is '
+                                        f'not allowed'}), 400
```
Comment (cmagnobarbosa) [inline main.py:112]: I think this `f'` (f-string) can be safely removed.

Comment (cmagnobarbosa) [inline main.py:112]: I think this `f'` (f-string) can be safely removed.

File: main.py [hunk 2]
```diff
@@ -90,6 +90,9 @@ def update_mw(self, mw_id):
         except KeyError:
             return jsonify({'response': f'Maintenance with id {mw_id} not '
                                         f'found'}), 404
+        if maintenance.status == Status.RUNNING:
+            return jsonify({'response': f'Updating a running maintenance is '
+                                        f'not allowed'}), 400
```
Comment (cmagnobarbosa) [inline main.py:95]: I think the `f-string` can be removed here

File: main.py (hunk 1 above)
Comment (hdiogenes) [inline main.py:112]:
```suggestion
            return jsonify({'response': 'Deleting a running maintenance is '
                                        'not allowed'}), 400
```

File: main.py (hunk 2 above)
Comment (hdiogenes) [inline main.py:95]:
```suggestion
            return jsonify({'response': 'Updating a running maintenance is '
                                        'not allowed'}), 400
```

File: main.py (hunk 1 above)
Comment (hdiogenes) [reply main.py]: Done. You can add the modifications as code suggestions so that it's easier to just accept+commit the changes, @cmagnobarbosa .

**Pull request merged**